### PR TITLE
Add user notification

### PR DESCRIPTION
**Pull request by WNjihia**
Closes issue #5 
When a data request is added, organization members are notified via email.

### CODE REVIEW

File: ckanext/datarequests/controllers/ui_controller.py [hunk 1]
```diff
@@ -174,7 +176,28 @@ def _process_post(self, action, context):
...
+                    extra_vars = {
+                        'site_title': config.get('ckan.site_title'),
+                        'site_url': config.get('ckan.site_url'),
+                        'datarequest_title': result['title'],
+                        'datarequest_description': result['description'],
+                        'datarequest_url': datarequest_url,
+                    }
+                    subject = base.render_jinja2('emails/notify_user_subject.txt',
+                                                 extra_vars)
+                    subject = subject.split('\n')[0]
```
Comment (DavidLemayian) [inline ckanext/datarequests/controllers/ui_controller.py]: Why are we splitting here?

File: ckanext/datarequests/controllers/ui_controller.py [hunk 2]
```diff
@@ -174,7 +176,28 @@ def _process_post(self, action, context):
...
+
+                    for user in users:
+                        user_data = model.User.get(user['id'])
+                        extra_vars['user_name'] = user_data.fullname
```
Comment (DavidLemayian) [inline ckanext/datarequests/controllers/ui_controller.py]: This should probably be called `user_fullname`

File: ckanext/datarequests/controllers/ui_controller.py (hunk 1 above)
Comment (WNjihia) [reply ckanext/datarequests/controllers/ui_controller.py]: Apologies. I have just realized that I don't need to split. I will make the necessary changes

File: ckanext/datarequests/controllers/ui_controller.py (hunk 2 above)
Comment (WNjihia) [reply ckanext/datarequests/controllers/ui_controller.py]: It's done. I have changed it to user_fullname

File: ckanext/datarequests/controllers/ui_controller.py (hunk 1 above)
Comment (WNjihia) [reply ckanext/datarequests/controllers/ui_controller.py]: Done

**Comment by DavidLemayian** [comment]
Thanks @WNjihia. Documentation needs to now be updated on `README.md`.

**Comment by DavidLemayian** [comment]
Let's also add any relevant doc strings.

**Comment by WNjihia** [comment]
@DavidLemayian Since the user will automatically be notified when a new data request is added, I don't think it will be necessary to add that information on the `README.md`

**Comment by DavidLemayian** [comment]
@WNjihia The adding of documentation isn't for users but for devs who re-use this plugin or are doing development on it. We'd like to show on the README.md the features that are available and how we've gone about implementation. e.g should they ensure SMTP configs are set in CKAN installation?

Having had to write that out has actually made me ask a few more questions:
1. Have we added test for this?
2. Is this the best place for this code to exist or should be referenced elsewhere instead? This is in fact `ui_controller` and perhaps this functionality should live in `actions.py` instead? Why the choice?

cc @shemogumbe

**Comment by WNjihia** [comment]
@DavidLemayian I have updated the README.md. I have also moved the function to `actions.py` and called it in the `ui_controller.py`.
Regarding the tests, could it be done as its own separate task?

**Comment by WNjihia** [comment]
@DavidLemayian I think this feature should be added in ckanext-notify. This is because, if merged to the ckanext-datarequests extension, every organization will be required to use email notifications whereas in ckanext-notify, they will be given an option of which notification channel to use

**Comment by DavidLemayian** [comment]
@WNjihia completely agree on all the points. This is awesome work. I'll close this and wait on your PR on ckanext-notify instead. :)

**Pull request closed without merging**